Merge upstream changes into registrator

chrisgilbert opened the pull request:
This PR merges the upstream changes that have occurred in gliderlabs/registrator.  There are couple of relevant fixes, and a useful change adding an `-Explicit` option.  This stops services being registered if they don't set the `SERVICE_REGISTER` environment var.

It also switches to using godep for dependencies.  It's an experiemental PR at the moment, and I want to see how easy it will be to merge these changes.

Commit:
Fix missing ca-certificates in alpine 3.5 (#523)

seriousben committed:
Fix existed typo (#539)

Shtsh committed:
Added support for DeregisterCriticalServiceAfter (#488)

Commit:
fix service ip and port in consul interpolateService (#453)

fix service ip and port in consul interpolateService

mattatcha committed:
upgrade circleci and vendor deps

mattatcha committed:
add setup remote docker step

mattatcha committed:
consul: set transport since HTTPClient may be nil

Sisir-Chowdhury committed:
Environment Variable name changed to CONSUL_CLIENT_CERT & CONSUL_CLIENT_KEY for TLS Cert/Pvt-Key

Sisir-Chowdhury committed:
Updated Docs for TLS Cert/Key Environment Variables

josegonzalez committed:
Merge pull request #566 from Sisir-Chowdhury/vendor

Environment Variable name changed to CONSUL_CLIENT_CERT & CONSUL_CLIE…

codablock committed:
Add option to only register services which have SERVICE_NAME (#531)

This adds the option "-explicit" which makes registrator ignore all
services which don't have a SERVICE_NAME label.

kichristensen committed:
Add Windows detection

avinson committed:
Improve CLI option documentation

avinson committed:
fix indent

josegonzalez committed:
Merge pull request #584 from avinson/master

Improve CLI option documentation

rifelpet committed:
Add link to Consul /agent/check/register docs

rifelpet committed:
Add docs for SERVICE_CHECK_DEREGISTER_AFTER

rifelpet committed:
Link to consul check docs from service model

rms1000watt committed:
Added Multistage Dockerfile for slimmer & hardened image

VeganBroccoli committed:
Add support for dep

dsouzajude committed:
Include service.Attrs as Consul Meta for the service

josegonzalez committed:
Merge pull request #615 from VeganBroccoli/buildfix

Add support for dep

josegonzalez committed:
Merge branch 'master' into MultistageDocker

josegonzalez committed:
Merge pull request #610 from rms1000watt/MultistageDocker

Added Multistage Dockerfile for slimmer & hardened image

josegonzalez committed:
Merge pull request #590 from rifelpet/docs

Consul check documentation improvement

josegonzalez committed:
Merge pull request #571 from kichristensen/WindowsDetection

Add Windows detection

josegonzalez committed:
feat: add circleci 2.0 support

josegonzalez committed:
Merge pull request #623 from gliderlabs/circleci

Add circleci 2.0 support

josegonzalez committed:
fix: do not use the global dockerignore for pagebuilder

Otherwise, the git directory is ignored.

josegonzalez committed:
Merge pull request #624 from gliderlabs/circle-test

hack: check circle directories

josegonzalez committed:
fix: Properly use CircleCI 2.0 for builds

The switch to circleci 2.0 completely broke page builds - amongst other things - as the docker build type doesn't support docker volumes.

Moving to the machine build type allows us to use docker volumes, though we lose the ability to use helper functions.

josegonzalez committed:
Merge pull request #625 from gliderlabs/circle-test

Properly use CircleCI 2.0 for builds

josegonzalez committed:
fix: add github.com to circleci known_hosts

josegonzalez committed:
fix: switch to non-eval command

This should fix doc builds.

dsouzajude committed:
Update docs for Consul metadata support

dsouzajude committed:
Use alpine:3.7 instead of scratch

josegonzalez committed:
Merge pull request #626 from dsouzajude/fix-build

Use alpine:3.7 instead of scratch

josegonzalez committed:
Merge pull request #620 from dsouzajude/consul-meta

Populate Consul's ServiceMeta using service.Attrs

Commit:
Dockerfile.dev - pull in changes made to main Dockerfile

Commit:
Dockerfile: fix whitespace formatting (space->tab)

josegonzalez committed:
Merge pull request #631 from jharley/fix-dev-dockerfile

Fix Dockerfile.dev

Commit:
Adds support for Consul's check http method

Commit:
Fixes documentation

eedwards-sk committed:
resolved merge conflicts

josegonzalez committed:
Merge pull request #658 from Snapkitchen/PR-568

resolve merge conflicts in PR-568 (update vendor libraries to fix tls support)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/hudl/registrator?pullRequest=56) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/hudl/registrator?pullRequest=56) before we can accept your contribution.<br/>**0** out of **15** committers have signed the CLA.<br/><br/>:x: mattatcha<br/>:x: seriousben<br/>:x: alfssobsd<br/>:x: Sisir-Chowdhury<br/>:x: Shtsh<br/>:x: josegonzalez<br/>:x: codablock<br/>:x: avinson<br/>:x: rifelpet<br/>:x: kichristensen<br/>:x: Jason Harley<br/>:x: rms1000watt<br/>:x: VeganBroccoli<br/>:x: dsouzajude<br/>:x: eedwards-sk<br/><hr/>**Jason Harley** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/hudl/registrator?pullRequest=56) it.</sub>

jamiegs commented:
If you merge this in would you also add https://github.com/hudl/registrator/pull/57?  It's too small of a change to warrant it's own deploy

nicofuccella committed:
Merge pull request #642 from dtm/feature/consul_http_method

Feature/consul http method

pmundt committed:
Convert from check.Script to check.Args for Consul API changes (#669)

In upstream Consul (as of May 2018), check.Script has been removed in
favour of check.Args, which now results in the following build error:

consul/consul.go:108:8: check.Script undefined (type *api.AgentServiceCheck has no field or method Script)
consul/consul.go:110:8: check.Script undefined (type *api.AgentServiceCheck has no field or method Script)
consul/consul.go:121:10: check.Script undefined (type *api.AgentServiceCheck has no field or method Script)

Update to use the new convention.

thesoftwarejedi committed:
Add consul grpc check

nicofuccella committed:
Merge pull request #672 from thesoftwarejedi/feature/consul-grpc-check

Add consul grpc check

psyhomb committed:
Support Go template in -tags flag

- Add custom Go template functions
- Update signiture of existing Go template fuctions to support pipelines
- Migrate to Go modules for managing dependencies
- Update versions of builder and runtime docker images (Dockerfile)
- Update README.md

testwill committed:
fix: information typo

josegonzalez committed:
Merge pull request #713 from testwill/typo

fix: information typo

starze committed:
Update README.md

Add CLI Option "explicit" do examples

josegonzalez committed:
Merge pull request #715 from starze/master

Update README.md

josegonzalez committed:
Merge pull request #677 from psyhomb/master

Add support for Go template to -tags flag
